### PR TITLE
add resubmit hook

### PR DESCRIPTION
**Pull request by malaffoon**
This is just a quick change so i can easily resubmit imports that get stuck in ACCEPTED state. That will (currently) happen under two conditions (both will be fixed but for now...):
1. There are no processors running so the rabbit queue has not been created
2. A processor was brought online with items in the queue; it seems to grab a couple and do nothing with them.

This hook can also be used to resubmit in the future when exams are "parked" because the assessment packages aren't available yet.

### CODE REVIEW

File: import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java [hunk 1]
```diff
@@ -81,4 +87,36 @@ public RdwImport importContent(final RdwUser user, final String payload, final I
...
+        if (query == null || query.isEmpty()) {
+            query = RdwImportQuery.builder()
+                    .status(ImportStatus.ACCEPTED)
+                    .after("-PT24H")
+                    .before("-PT1H")
+                    .build();
+        }
+        final Map<String, TenancyChain> userTenancyChains = newHashMap();
+        long count = 0;
+        for (final RdwImport rdwImport : repository.findBy(query)) {
```
Comment (malaffoon) [inline import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java]: In the final version we may want to make all this asynchronous and handle the query in chunks or something.

File: import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java [hunk 2]
```diff
@@ -81,4 +87,36 @@ public RdwImport importContent(final RdwUser user, final String payload, final I
...
+            final Properties properties = archiveService.readProperties(location);
+            final String contentType = properties.getProperty(ContentTypeProperty);
+            final String username = properties.getProperty(UsernameProperty);
+            // TODO - fetch current tenancy chain for user from authz service
+            // TODO   skipping for now because authz is changing Real Soon Now ...
+            final TenancyChain tenancyChain = userTenancyChains.computeIfAbsent(username, k ->
+                    TenancyChain.fromString(properties.getProperty(TenancyChainProperty)));
+
+            source.submitContent(username, tenancyChain, payload, content, contentType, rdwImport.getId());
+            ++count;
```
Comment (bgsavage) [inline import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java]: do we need to keep count, or can we just return the length of the list coming from repository.findby ?

Comment (malaffoon) [inline import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java]: You're right, for now we could return the length of the list. The count is left over from some chunking experimentation i was doing.

File: import-service/src/main/java/org/opentestsystem/rdw/ingest/web/ExamController.java [hunk 3]
```diff
@@ -89,4 +90,17 @@ public RdwImportCsv getImportsAsCsv(final RdwImportQuery query) {
...
+    @PostMapping("/imports/resubmit")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    public ResponseEntity<Long> resubmitImports(final RdwImportQuery query) {
+        // if query is empty, force to be ACCEPTED
+        return ResponseEntity.ok(service.resubmitImports(query.isEmpty() ?
+                RdwImportQuery.builder().content(ImportContent.EXAM).status(ImportStatus.ACCEPTED).build() :
+                ensureExamQuery(query)));
+    }
+
+    private static RdwImportQuery ensureExamQuery(final RdwImportQuery query) {
```
Comment (bgsavage) [inline import-service/src/main/java/org/opentestsystem/rdw/ingest/web/ExamController.java]: I think the naming of this confused me.  I asked "ensure Exam query what?".  Maybe ensureQueryIncludesExam is clearer?

Comment (malaffoon) [inline import-service/src/main/java/org/opentestsystem/rdw/ingest/web/ExamController.java]: will change

**Pull request merged**